data/scripts.ts

```diff
@@ -1013,7 +1013,7 @@ export const BattleScripts: BattleScriptsData = {
 		}
 
 
-		if (!didAnything && didAnything !== 0 && !moveData.self && !moveData.selfdestruct) {
+		if (!didAnything && didAnything !== 0 && !moveData.self && !moveData.selfDamage?.startsWith('ko')) {
```

Comment:
They all start with `'KO'` not `'ko'`, right? Is this a bug?

Reply:
I do think that if you're doing substring searches, this should probably be its own field instead of rolled into `selfDamage`, though.